instrumentctl/g9_driver.py

```diff
@@ -0,0 +1,254 @@
+import serial
+import threading
+import time
+from utils import LogLevel
+import os
+from subsystem import interlocks
+
+inStatus = {
+    0: "No error",
+    1: "Invalid configuration",
+    2: 'External test signal failure',
+    3: 'Internal circuit error',
+    4: 'Discrepancy error',
+    5: 'Failure of the associated dual-channel input'
+}
+
+outStatus = {
+    0: 'No error',
+    1: 'Invalid configuration',
+    2: 'Overcurrent detection',
+    3: 'Short circuit detection',
+    4: 'Stuck-at-high detection',
+    5: 'Failure of the associated dual-channel output',
+    6: 'Internal circuit error',
+    8: 'Dual channel violation'
+}
+
+usStatus = {
+    9: "Output Power Supply Error Flag",
+    10: "Safety I/O Terminal Error Flag",
+    13: "Function Block Error Flag"
+}
+
+class G9Driver:
+    def __init__(self, port=None, baudrate=9600, timeout=0.5, logger=None, debug_mode=False):
+        if port:
+            self.ser = serial.Serial(port, baudrate, parity=serial.PARITY_EVEN, stopbits=serial.STOPBITS_ONE, bytesize=serial.EIGHTBITS, timeout=timeout)
+        self.debug_mode = debug_mode
+        self.logger = logger
+        self.lastResponse = None
+        self.msgOptData = None
+
+    def send_command(self):
+        if not self.is_connected():
+            raise ConnectionError("Seiral Port is Not Open.")
+        header = b'\x40\x00\x00\x0F\x4B\x03\x4D\x00\x01' 
+        data = b'\x00\x00\x00\x00\x00\x00'
+        self.msgOptData = data
+        checksum = self.calculate_checksum(header + data, 0 , len(header + data))
+        cs = b'\x00' + checksum if len(checksum) == 1 else checksum
+        footer = b'\x2A\x0D' 
+        self.ser(header + data + cs + footer)
```

Comment:
I don't think `self.ser` can be called with this data. Probably has to be `self.ser.write(header + data + cs + footer)` instead.